### PR TITLE
Initial azure health implementation

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/giantswarm/issues/6893

### CODE REVIEW

File: service/health/searcher/azure.go [hunk 1]
```diff
@@ -2,12 +2,33 @@ package searcher
...
+	azureCR, err := s.g8sClient.
+		ProviderV1alpha1().
+		AzureConfigs("default").
+		Get(request.ClusterID, v1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	var health string
+	{
+		nodeCount := len(azureCR.Status.Cluster.Nodes)
```
Comment (MarcelMue) [inline service/health/searcher/azure.go]: Fine as the very first implementation. We should start thinking about how we split this logic. 
The logic for health calculation is pretty much the same for all providers - so how can we reuse this?

File: mock/azure.go [hunk 2]
```diff
@@ -1,4 +1,210 @@
 package mock
 
 // AzureHealthy is a JSON string representing a healthy Azure cluster
-var AzureHealthy = `{}`
+var AzureHealthy = `
```
Comment (MarcelMue) [inline mock/azure.go]: Does this mock get used right away now?

**Review by MarcelMue — APPROVED** [review]
Fine with me if we iterate on it soon.

File: mock/azure.go (hunk 2 above)
Comment (tfussell) [reply mock/azure.go]: Yes, all wired up! Checked by temporarily removing nodes and getting `red` health as expected.

File: service/health/searcher/azure.go (hunk 1 above)
Comment (tfussell) [reply service/health/searcher/azure.go]: This is what I'm working on now. Will discuss with them team once I have some more concrete ideas.

**Pull request merged**